gui.py

```diff
@@ -1,157 +1,219 @@
-    spec_lbl.place(x=spec_lbl_x, y=spec_lbl_y)
-    root.update()
-
-    spec_lbl_dims = get_lbl_dims(spec_lbl)
-    spec_combobox = ttk.Combobox(root, values=["Vref", "option 2"])
-    spec_combobox.place(
-        x=spec_lbl_dims.x + spec_lbl_dims.width + 20,
-        y=spec_lbl_dims.y
-    )
-
-    spec_combobox.current(0)
-
-    # place dataset label ---------------------------------------------------------------------------------------------
-    dataset_lbl = tk.Label(root, text="Choose a dataset:")
-    dataset_lbl_x = spec_lbl_dims.x
-    dataset_lbl_y = spec_lbl_dims.y + spec_lbl_dims.height + 20
-    dataset_lbl.place(x=dataset_lbl_x, y=dataset_lbl_y)
-    root.update()
-
-    dataset_lbl_dims = get_lbl_dims(dataset_lbl)
-    dataset_combobox = ttk.Combobox(root, values=["25", "option 2"])
-    dataset_combobox.place(
-        x=dataset_lbl_dims.x + dataset_lbl_dims.width + 20,
-        y=dataset_lbl_dims.y
-    )
-
-    dataset_combobox.current(0)
+import csv
+
+# Create Window for GUI
+root = tk.Tk()
+
+# Set title of the window
+root.title("Radiation on BJT explorer")
+
+# Set dimensions of the window
+root.geometry("800x750")
+
+# Set the background color of the root window
+root.configure(bg="darkgray")
+
+# Adding frames for parts and specifications
+def create_frame(row, column, text, width=2, height=2):
+    frame = LabelFrame(root, text=text, padx=10, pady=10, borderwidth=2, relief="solid", width=width, height=height)
+    frame.grid(row=row, column=column, padx=10, pady=10, columnspan=width, rowspan=height, sticky="ewns")
+
+    return frame
+
+# Validation function to allow only numerical values
+def validate_numerical(value):
+    if value == '':
+        return True
+    try:
+        float(value)
+        return True
+    except ValueError:
+        return False
+
+# Backend function
+def execute_backend():
+
+    # proper variable names
+    Voltage = textbox_dataset_vcc.get()
+    Temperature = textbox_temp.get()
+    Fluence_Min = textbox_fluences_min.get()
+    Fluence_Max = textbox_fluences_max.get()
+
+    # Write values to a file
+    with open('shared-config/shared_values.txt', 'w') as file:
+        file.write(f"Voltage: {Voltage}\n")
+        file.write(f"Temperature: {Temperature}\n")
+        file.write(f"Fluence Min: {Fluence_Min}\n")
+        file.write(f"Fluence Max: {Fluence_Max}\n")
+
+    # Build the command to run backend.py
+    command = ["python", "backend.py"]
+    # Run backend.py as a separate process
+    subprocess.run(command)
+    
+    # Validate the input before converting to float
+    if Fluence_Min and Fluence_Max:
+        range_min = float(Fluence_Min) * 10**11
+        range_max = float(Fluence_Max)  * 10**13
+        print(range_min, range_max)
+        x_range_to_include = (range_min , range_max)
+        execute_function(sample_data, x_range_to_include)
+    else:
+        execute_function(sample_data)
+
+# Execute function
+def execute_function(sample_data, x_range=None):
```

Comment:
Can you leave a docstring or comment saying what this function does? The name does not tell me much.